docs/data-deduplication.md

```diff
@@ -0,0 +1,121 @@
+- Pro: higher data deduplication rate.
+- Pro: friend to on demand loading
+- Pro: friend to CDN for data cache.
+- Con: not friend to Container Registry.
+- Con: not friend to random access.
+- Con: higher CPU/memory resource consumption when accessing the deduplicated data.
+
+In a backup program, data de-duplication can be applied in two locations:
+Removing duplicate data from the same or different files within the same backup process (inter-file de-duplication), e.g. during the initial backup,
+or removing it between several backups that contain some of the same data (inter-backup de-duplication).
+While the former is desirable to have, the latter is much more important.
+So CDC is suitable/optimized for data backup.
+
+## Related Projects
+There are several related projects which try to reduce duplicated data in container image/software package by using data deduplication.
+
+### casync - Content Addressable Data Synchronizer
+[casync](https://github.com/systemd/casync) takes inspiration from the popular rsync file synchronization tool as well as the probably even more popular git revision control system.
+It combines the idea of the rsync algorithm with the idea of git-style content-addressable file systems, and creates a new system for efficiently storing and delivering file system images, optimized for high-frequency update cycles over the Internet.
+Its current focus is on delivering IoT, container, VM, application, portable service or OS images.
+
+### CernVM-FS
+The [CernVM File System](https://cernvm.cern.ch/fs/) provides a scalable, reliable and low-maintenance software distribution service.
+It was developed to assist High Energy Physics (HEP) collaborations to deploy software on the worldwide-distributed computing infrastructure used to run data processing applications.
+CernVM-FS is implemented as a POSIX read-only file system in user space (a FUSE module).
+Files and directories are hosted on standard web servers and mounted in the universal namespace /cvmfs.
+
+Internally, CernVM-FS uses content-addressable storage and Merkle trees in order to store file data and meta-data.
+CernVM-FS uses outgoing HTTP connections only, thereby it avoids most of the firewall issues of other network file systems.
+It transfers data and meta-data on demand and verifies data integrity by cryptographic hashes.
+
+### OSTree
+[OSTree](https://github.com/ostreedev/ostree) is an upgrade system for Linux-based operating systems that performs atomic upgrades of complete filesystem trees.
+It is not a package system; rather, it is intended to complement them.
+A primary model is composing packages on a server, and then replicating them to clients.
+
+The underlying architecture might be summarized as “git for operating system binaries”.
+It operates in userspace, and will work on top of any Linux filesystem.
+At its core is a git-like content-addressed object store with branches (or “refs”) to track meaningful filesystem trees within the store.
```

Comment:
nope, I don't think Ostree is strictly git-like system now, since git already has packfile and gc features. we shouldn't clarify git as ostree-like system.

Reply:
It's quotation from ostree project doc：）

Reply:
I'd suggest we shouldn't quote them directly but reflect the truth.

Reply:
see https://git-scm.com/docs/git-pack-objects

A packed archive is `an efficient way to transfer a set of objects between two repositories` as well as `an access efficient archival format`.

Reply:
please append some words like:

however, git also introduced packed archive feature since storing `git object` isn't quite effective, also see [git-pack-objects](https://git-scm.com/docs/git-pack-objects)

Reply:
Refined the descriptions about related projects.